R/tm_g_ci.R

```
@@ -407,12 +408,11 @@ srv_g_ci <- function(id, # nolint
       selector_list = selector_list
     )
 
-    anl_q <- shiny::reactive(
-      teal.code::eval_code(
-        object = teal.code::new_qenv(tdata2env(data), code = get_code_tdata(data)),
-        code = as.expression(anl_inputs()$expr)
-      )
-    )
+    anl_q <- shiny::reactive({
+      req(anl_inputs())
```

why req() ?

---

It's always safer to add `req()` of the reactive/input that you use inside the reactive expression. Isn't it?

---

I saw some places use it, some places not. In this case if `anl_inputs` is `NULL` nothing meaningful happens, so it makes sense to add it.

---

In place its used  variable contains user inputs.

---

I didn't get you. Could you explain what you mean?

---

Wait, be careful with the `req`s. You might mask validation messages.
Also, please keep to the scope of the issue, we don't want to loose track of this beast.

---

I  believed that the variable `anl_q` did not store user input, leading me to think it didn't req(). never mind

---

It's a common misconception that `req()` should only be used with inputs. But they can be used with reactives too.

---

@vedhav Please make sure all input validation messages are intact after adding this `req`. You may have to compare to `main`. If it's actually needed for something, please raise a separate issue. If it wasn't, why add it?

---

I noticed some silent errors in some modules. For example in tmg's `tm_missing_data` which you can [observe in the ADLB tab of the exploratory app](https://genentech.shinyapps.io/nest_exploratory_dev/) of course this won't fix it but felt a need to increase the usage of `req()`.

---

Good. Separate issue, please.

---

I'll create it in `teal.gallery`

---

Do you prefer to revert this `req()` or leave the poor guy alone?

---

**If** you confirm it didn't break anything, leave it in. Just be thorough.

---

I'm certain it's not breaking anything. My ocd kicked in. We have nice module specific PRs just for this issue and I don't want to taint it :D